pcdsdevices/lasers/qmini.py

```diff
@@ -83,6 +87,47 @@ class QminiSpectrometer(Device):
     fit_stdev = Cpt(EpicsSignalRO, ':STDEV', kind='config')
     fit_chisq = Cpt(EpicsSignalRO, ':CHISQ', kind='config')
 
+    # Save spectra functions
+    def save_data(self):
+        """
+        Save the wavelength and spectrum PVs to a text file
+        """
+        if len(self.file_dest.get().strip()) == 0:
+            # set a default destination for the file if you're lazy or give it whitespace
+            _file = os.getcwd() + '/' + self.name + time.strftime("_%Y-%m-%d_%H%M%S") + '.txt'
+        else:
+            _file = self.file_dest.get()
+        self.log.info('Saving spectrum to disk...')
+        # Let's format to JSON for the science folk with sinful f-string mangling
+        _settings = ['sensitivity_cal', 'correct_prnu', 'correct_nonlinearity',
+                     'normalize_exposure', 'adjust_offset', 'subtract_dark',
+                     'remove_bad_pixels', 'remove_temp_bad_pixels']
+        _data = {'timestamp': time.strftime("%Y-%m-%d %H:%M:%S"),
+                 'exposure (us)': str(self.exposure.get()),
+                 'averages': str(self.exposures_to_average.get()),
+                 # Lets do some sneaky conversion to bool from int
+                 'settings': {f"{sig}": str(eval(f"{self}.{sig}.get() > 0")).lower()
```

Comment:
Nitpick: maybe simpler without an `eval` so the intent is clearer and the next maintainer doesn't get confused.
I probably botched this because I also got confused. I left it as an int because ints are allowed in json.
```suggestion
                 'settings': {f"{sig}": int(getattr(self, sig).get())
```

Reply:
I neglect `getattr` a little too often for my own good, that's much cleaner. `int` are allowed but I preferred `true` and `false` outputs for the states to match their respective PVs as `bo` records. 

This is maybe just convenience for the consumer of the dump, since it saves them a comparison when checking settings in the object.

I might just be enforcing my own preferences, tbh, so it isn't strictly necessary. Maybe I'll ask the scientists what they prefer 🤷

Reply:
Gotcha, I guess since the point is for the end-user to read there could a lot of ways to do it!

I you can also put boolean literals into json and it looks like `true` or `false` without the quotes.

Reply:
I guess I could've just cast to `bool` anyway, right? Not sure why I compared to get there in the first place lol. Very good point, I like that much better and will do that